### PR TITLE
add test for invalid_tag

### LINKED ISSUE #69: Test invalid_tag? in record_level.rb
Acceptance criteria:
- [x] Fixture has a field with a non-numeric tag
- [x] Test detects the non-numeric field
- [x] All lines of the `invalid_tag?` method have test coverage

### PR DESCRIPTION
**Pull request by pmgreen**
Closes #69

### CODE REVIEW

File: spec/record_level/validate_marc_spec.rb [hunk 1]
```diff
@@ -56,13 +56,19 @@
...
                      'indicator2' => ' ',
-                     'subfields' => [{ 'a' => 'Invalid field' }] } }
+                     'subfields' => [{ 'a' => 'Invalid field' }] } },
+        { 'ABC' => { 'indicator1' => ' ',
+                     'indicator2' => ' ',
+                     'subfields' => [{ 'a' => 'Non-numeric tag' }] } }
       ]
     end
     let(:record) { MARC::Record.new_from_hash('fields' => fields) }
     it 'finds the invalid tag' do
```
Comment (Beck-Davis) [inline spec/record_level/validate_marc_spec.rb:66]: Not sure what this block is doing. The code only checks for non-numerical characters in the field tag. This block isn't really necessary.

**Review by Beck-Davis — CHANGES_REQUESTED** [review]
Separate methods should have separate test blocks under the `RSpec.describe`. We should avoid nesting tests in unrelated methods (they don't call on each other).

**Comment by pmgreen** [comment]
Ok, I wasn't sure whether validate_marc was the method in record_level.rb or whether it was the logical structure for tests.

**Comment by Beck-Davis** [comment]
I double checked, it looks like we've been creating separate files for each method to be tested, even if the method is smaller, unless they are directly related or one calls the other. So for this one, it can have its own file.

**Comment by pmgreen** [comment]
The `invalid_tag?` method is currently only checking for non-numeric characters within a tag (not length, etc.). Maybe rename it to `non_numeric_tag?`.

**Review by Beck-Davis — APPROVED** [review]
looks good

**Comment by Beck-Davis** [comment]
> The `invalid_tag?` method is currently only checking for non-numeric characters within a tag (not length, etc.). Maybe rename it to `non_numeric_tag?`.

True. We can make a ticket for it.

**Pull request merged**